catalog/content/core/content-core-catalogerplugin/src/main/java/ddf/content/plugin/cataloger/CatalogContentPlugin.java

```diff
@@ -280,18 +276,63 @@ private Metacard generateMetacard(ContentItem contentItem, MimeType mimeType, St
                         LOGGER.debug("Unable to retrieve Security Manager.", e);
                     }
 
-                    if (uri != null) {
-                        //Setting the non-transformer specific information not including creation and modification dates/times
-                        contentMetacard.setAttribute(new AttributeImpl(Metacard.RESOURCE_URI, uri));
-                        contentMetacard.setAttribute(new AttributeImpl(Metacard.RESOURCE_SIZE,
-                                String.valueOf(size)));
+                    if (generatedMetacard instanceof DynamicMetacard) {
+                        LOGGER.debug("Adding fields to dynamic metacard of type {}", generatedMetacard.getContentTypeName());
+                        DynamicMetacard metacard = (DynamicMetacard) generatedMetacard;
+
+                        // set the user name
+                        metacard.setAttribute(Metacard.POINT_OF_CONTACT, name);
+
+                        // set the uri
+                        if (uri != null) {
+                            metacard.setAttribute(Metacard.RESOURCE_URI, uri);
+                            metacard.setAttribute(Metacard.RESOURCE_SIZE, String.valueOf(size));
+                        } else {
+                            LOGGER.debug("Metacard had a null uri");
+                        }
+                        if (StringUtils.isBlank(metacard.getTitle())) {
+                            LOGGER.debug("Metacard title was blank. Setting title to filename.");
+                            metacard.setAttribute(Metacard.TITLE, contentItem.getFilename());
+                        }
+                        contentMetacard = generatedMetacard;
```

Comment:
It was being pulled out above the if/else check and then the POINT_OF_CONTACT set with the name variable. Changed the else block to do the same.